internal/pkg/util/bin/bin.go

```diff
@@ -33,28 +33,42 @@ func FindBin(name string) (path string, err error) {
 		return findOnPath(name)
 	// Configurable executables that are found at build time, can be overridden
 	// in apptainer.conf. If config value is "" will look on PATH.
-	case "unsquashfs", "mksquashfs", "go":
+	case "unsquashfs", "mksquashfs", "go", "cryptsetup", "ldconfig", "nvidia-container-cli":
 		return findFromConfigOrPath(name)
 	// distro provided setUID executables that are used in the fakeroot flow to setup subuid/subgid mappings
 	case "newuidmap", "newgidmap":
 		return findOnPath(name)
-	// cryptsetup & nvidia-container-cli paths must be explicitly specified
-	// They are called as root from the RPC server in a setuid install, so this
-	// limits to sysadmin controlled paths.
-	// ldconfig is invoked by nvidia-container-cli, so must be trusted also.
-	case "cryptsetup", "ldconfig", "nvidia-container-cli":
-		return findFromConfigOnly(name)
 	}
 	return "", fmt.Errorf("unknown executable name %q", name)
 }
 
-// findOnPath performs a simple search on PATH for the named executable, returning its full path.
-// env.DefaultPath` is appended to PATH to ensure standard locations are searched. This
-// is necessary as some distributions don't include sbin on user PATH etc.
+// findOnPath performs a search on the configurated binary path for the
+// named executable, returning its full path.
 func findOnPath(name string) (path string, err error) {
+	cfg := apptainerconf.GetCurrentConfig()
+	if cfg == nil {
+		if strings.HasSuffix(os.Args[0], ".test") {
```

Comment:
Instead of having this directly in the code, I would put the parsing and initialization of current config directly in `_test.go` files

Reply:
The problem with doing that here is that this function is used from other unit tests as well via the FindBin function, and it would have to be included in multiple `_test.go` files if it wasn't here.  So I think the loading config files for .test in this file need to stay here.